src/main/java/seedu/address/model/module/NusModsShareLink.java

```diff
@@ -0,0 +1,99 @@
+package seedu.address.model.module;
+
+import static java.util.Map.entry;
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import seedu.address.logic.parser.ParserUtil;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.websocket.util.UrlUtil;
+
+/**
+ * Represents an NUSMods share link.
+ */
+public class NusModsShareLink {
+
+    public static final String EXAMPLE = "https://nusmods.com/timetable/sem-1/share?CS2101=&CS2103T=LEC:G05&CS3230"
```

Comment:
I like that a sample link is present, can be very useful for writing test cases :+1: